src/core_plugins/kibana/public/management/react/store/actions/index_pattern.js

```diff
@@ -0,0 +1,67 @@
+import { globals } from '../../globals';
+import { createAction } from 'redux-actions';
+import { createThunk } from 'redux-thunks';
+
+export const fetchedIndexPattern = createAction('FETCHED_INDEX_PATTERN',
+  (pattern, fields, timeFieldName, isDefault, id) => ({ pattern, fields, timeFieldName, isDefault, id }));
+export const fetchIndexPattern = createThunk('FETCH_INDEX_PATTERN',
+  async ({ dispatch }, pattern) => {
+    const rawIndexPattern = await globals.es.search({
+      index: '.kibana',
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                match: {
+                  type: 'index-pattern',
+                },
+              },
+              {
+                match: {
+                  'index-pattern.title': pattern,
+                }
+              }
+            ]
+          }
+        }
+      }
+    });
+    const hits = rawIndexPattern.hits.hits;
+    if (hits.length !== 1) {
+      throw 'Index pattern query did not match 1 index pattern. This is just debug for now';
+    }
+    const { _id, _source: source } = hits[0];
+    const id = _id.replace('index-pattern:', '');
+    const details = source['index-pattern'];
+    const timeFieldName = details.timeFieldName;
+    const fields = details.fields ? JSON.parse(details.fields) : [];
+    const isDefault = id === globals.config.get('defaultIndex');
+    dispatch(fetchedIndexPattern(pattern, fields, timeFieldName, isDefault, id));
```

Comment:
The use of `fetch` and `fetched` here is a little awkward. There's actually a shortcut here since you're using a thunk...

`FETCH_INDEX_PATTERN` is not an action, it's just a function you call that dispatches another action. That means you can dispatch an action internally that has the same name (and in fact, redux-thunks even provides a name property to make this easier) and use the same action name in the reducer.

```js
export const fetchIndexPattern = createThunk('FETCH_INDEX_PATTERN', ({ dispatch, type }, pattern) => {
  // ... code as above, async and whatever

  // dispatch a real action with a matching action type (provided by the `type` param in the first argument object)
  const action = createAction(type);
  dispatch(action({ pattern, fields, timeFieldName, isDefault, id }));
});
```

Now in your reducer you can simply `import { fetchIndexPattern } from '../path/to/action'` and it'll get used because the `toString` values match, and you no longer have to juggle the thunk and the actual action.